### PR TITLE
Add Universal ES6 Compiling

### PR DESCRIPTION
**Pull request by Serubin**
Previously there was no prepossessing for vue-h-ajax that would enable universal ES6 compiling. This should add that - tested on Vue webpack config and Vue browserify config using npm pack.

### CODE REVIEW

**Comment by Serubin** [comment]
@zont I tested the previous master commit 1c826bf using jest and it also failed.

**Comment by zont** [comment]
Travis check ok: https://travis-ci.org/zont/vue-h-ajax/builds/282293658?utm_source=github_status&utm_medium=notification

**Comment by Serubin** [comment]
I merged in your latest head. My code doesn't actually touch the module's code - could you help track down the jest issue?

File: package.json [hunk 1]
```diff
@@ -2,14 +2,18 @@
...
+  "module": "dist/index.js",
   "keywords": [
     "vue",
     "http",
     "ajax",
     "xhr"
   ],
   "scripts": {
+    "prebuild": "eslint index.js",
+    "build": "babel index.js --out-file dist/index.js",
```
Comment (zont) [inline package.json]: babel is called, but not present in devDependencies

File: .babelrc [hunk 2]
```diff
@@ -1,5 +1,10 @@
 {
   "presets": [
-    "es2015"
+    [
+      "es2015",
+      {
+        "modules": false
```
Comment (zont) [inline .babelrc]: This line breaks the tests.
This line breaks usage vue-h-ajax as module.

File: package.json [hunk 3]
```diff
@@ -22,21 +26,21 @@
...
+  },
   "author": "Alexander Zonov",
   "license": "MIT",
   "dependencies": {},
   "devDependencies": {
     "babel-jest": "^18.0.0",
     "babel-preset-es2015": "^6.22.0",
     "eslint": "^3.15.0",
-    "jest": "^18.1.0",
+    "jest": "^21.2.1",
```
Comment (zont) [inline package.json]: All tests are checked using jest 18.
Why you change jest version?
Is version 21 full compatible with 18?

**Comment by Serubin** [comment]
Thanks for the guidance. Resolved all the issues.

**Comment by zont** [comment]
Build is failed if dist folder not existed

File: .babelrc [hunk 4]
```diff
@@ -1,5 +1,7 @@
 {
   "presets": [
-    "es2015"
+    [
+      "es2015",
```
Comment (zont) [inline .babelrc]: If this equals to "presets": ["es2015"] then it no need for pull request.
Chenges should be minimal.

File: package.json [hunk 5]
```diff
@@ -22,21 +26,23 @@
       "html"
     ]
   },
+  "browserify": {
```
Comment (zont) [inline package.json]: Changes should be minimal.
No need to move section to another place in the file

**Comment by zont** [comment]
As docs say (https://docs.npmjs.com/misc/scripts):
`
prepare: Run both BEFORE the package is packed and published, and on local npm install without any arguments
`

May be not includes /dist in .gitignore and use prepublishOnly for make build only before publish?
It makes install more quick.

**Comment by Serubin** [comment]
Up to you. Personal preference is too keep generated files like that out of the repo and generate on download. My reasoning is that you may make a change to the source and not update generated code creating an inconsistency. That makes using `prepare` make more sense.

Alternatively, we can keep `dist/` in `.gitignore` and use `prepublishOnly` to keep the install time through npm down.  Does that sound okay?

**Comment by zont** [comment]
I prefer the use of prepublishOnly script.

**Comment by Serubin** [comment]
Changes made.

**Comment by zont** [comment]
dist/* in .gitignore makes prepublish build meaningless.

**Comment by Serubin** [comment]
It shouldn't effect it. You can test using npm pack. It'll create a tar
archive with what would be uploaded to npm.

**Comment by zont** [comment]
Yes, I was wrong.

**Pull request merged**